rdt: change the name of the root group/class to DEFAULT

marquiz opened the pull request:
Change slightly obscure SYSTEM_DEFAULT to just DEFAULT.

Also, add an alias so that an empty string can also be used instead of
"DEFAULT" to refer to the root class. This should make the API more
consistent.

askervin commented:
+1 for the alias.

An empty string as an RDT class name in the configuration looks cryptic. And so would do any special character ("/" or "*", for instance). In my opinion `root` is pretty good, and so would be `default`. RDT documentation says:

> Resource groups are represented as directories in the resctrl file
> system.  The *default* group is the *root* directory which, immediately
> after mounting, owns all the tasks and cpus in the system and can make
> full use of all resources.

Therefore, I'd be tempted to favor `default` for two reasons:

1. It is a group name and does not refer to directory structure.

2. The same name would work in the blockio class configuration, too, if we'd like to allow user to define default blockio settings for containers without any other class.

marquiz commented:
> Therefore, I'd be tempted to favor `default` for two reasons:
> 
> 1. It is a group name and does not refer to directory structure.
> 2. The same name would work in the blockio class configuration, too, if we'd like to allow user to define default blockio settings for containers without any other class.

Mm, I'm inclined to agree.

I'd probably use capitalized version (`DEFAULT`) to try to avoid naming clashes. Some users might have `default` class (other than the resctrl root). This naming convention would make the uniqueness of the resctrl root a bit more obvious. WDYT?

askervin commented:
> Some users might have default class (other than the resctrl root).

Good point. I agree. `DEFAULT` is good. I'll add that to the blockio, too.

marquiz commented:
> > Some users might have default class (other than the resctrl root).
> 
> Good point. I agree. `DEFAULT` is good. I'll add that to the blockio, too.

👍  Changed the patch accordingly. Implementing the alias proved to be surprisingly simple – neater than I thought (why I added that as TODO item in the first iteration). I changed it so that `DEFAULT` is the "real name" and an empty string is handled as an alias. This way e.g. log printouts are more readable as we see DEFAULT instead of an empty string.

askervin reviewed (approved):
Very nice.
